builder/vsphere/driver/vm.go

```diff
@@ -509,41 +515,51 @@ func addDisk(_ *Driver, devices object.VirtualDeviceList, config *CreateConfig)
 
 func addNetwork(d *Driver, devices object.VirtualDeviceList, config *CreateConfig) (object.VirtualDeviceList, error) {
 	var network object.NetworkReference
-	if config.Network == "" {
-		h, err := d.FindHost(config.Host)
-		if err != nil {
-			return nil, err
+	for _, nic := range config.NICs {
+		if nic.Network == "" {
+			h, err := d.FindHost(config.Host)
+			if err != nil {
+				return nil, err
+			}
+
+			i, err := h.Info("network")
+			if err != nil {
+				return nil, err
+			}
+
+			if len(i.Network) > 1 {
+				return nil, fmt.Errorf("Host has multiple networks. Specify it explicitly")
+			}
+
+			network = object.NewNetwork(d.client.Client, i.Network[0])
```

Comment:
I'm wondering if this should run if  `len(config.NICs) == 0 && nic.Network == ""`.  
`config.NICs` can be empty in case the person doesn't set any Network adapters and a Netword card, right?

Reply:
We would need to identify a default network adapter type (networkCard). They didn't have this before so I didn't attempt to change this behavior. The empty string network also only works if there is only one network on the host. I don't think that ends up being very useful.

Reply:
How about adding an error if no network adapters are defined as Packer would need at least one to get past the "waiting for IP"?